app/classes/query/modules/ordering.rb

```diff
@@ -1,121 +1,219 @@
-    columns = model.column_names
-    case by
+        "#{model.table_name}.last_login DESC"
+      end
 
-    when "updated_at", "created_at", "last_login", "num_views"
-      "#{table}.#{by} DESC" if columns.include?(by)
+      def sort_by_num_views(model)
+        return unless model.column_names.include?("num_views")
+
+        "#{model.table_name}.num_views DESC"
+      end
+
+      def sort_by_date(model)
+        if model.column_names.include?("date")
+          "#{model.table_name}.date DESC"
+        elsif model.column_names.include?("when")
+          "#{model.table_name}.when DESC"
+        elsif model.column_names.include?("created_at")
+          "#{model.table_name}.created_at DESC"
+        end
+      end
 
-    when "date"
-      if columns.include?("date")
-        "#{table}.date DESC"
-      elsif columns.include?("when")
-        "#{table}.when DESC"
-      elsif columns.include?("created_at")
-        "#{table}.created_at DESC"
+      def sort_by_name(model)
+        sort_by_name_method = "sort_#{model.name.underscore.pluralize}_by_name"
+        if ::Query::Modules::Ordering.private_method_defined?(
+          sort_by_name_method
+        )
+          send(sort_by_name_method)
+        else
+          sort_other_models_by_name(model)
+        end
       end
 
-    when "name"
-      if model == Image
+      def sort_images_by_name
         add_join(:observation_images, :observations)
         add_join(:observations, :names)
         self.group = "images.id"
         "MIN(names.sort_name) ASC, images.when DESC"
-      elsif model == Location
+      end
+
+      def sort_locations_by_name
         if User.current_location_format == "scientific"
           "locations.scientific_name ASC"
         else
           "locations.name ASC"
         end
-      elsif model == LocationDescription
+      end
+
+      def sort_location_descriptions_by_name
         add_join(:locations)
         "locations.name ASC, location_descriptions.created_at ASC"
-      elsif model == Name
+      end
+
+      def sort_names_by_name
         "names.sort_name ASC"
-      elsif model == NameDescription
+      end
+
+      def sort_name_descriptions_by_name
         add_join(:names)
         "names.sort_name ASC, name_descriptions.created_at ASC"
-      elsif model == Observation
+      end
+
+      def sort_observations_by_name
         add_join(:names)
         "names.sort_name ASC, observations.when DESC"
-      elsif columns.include?("sort_name")
-        "#{table}.sort_name ASC"
-      elsif columns.include?("name")
-        "#{table}.name ASC"
-      elsif columns.include?("title")
-        "#{table}.title ASC"
       end
 
-    when "title", "login", "summary", "copyright_holder", "where",
-         "initial_det", "accession_number"
-      "#{table}.#{by} ASC" if columns.include?(by)
+      def sort_other_models_by_name(model)
+        if model.column_names.include?("sort_name")
+          "#{model.table_name}.sort_name ASC"
```

Comment:
@pellaea, @nimmolo 
This line is untested, and I think it's unreachable in practice because:
- There are only two tables with a `sort_name` column:
`names` and `names_version`.
- `names` is already handled by `sort_names_by_name`
-  We don't sort `names_version` by `sort_name`.

**Should I delete the first condidtion?**

Reply:
Don't know enough to have an opinion here.

Reply:
Sounds right, rip it out.